views/js/pciManager/pciManager.js

```diff
@@ -82,180 +84,231 @@ define([
+            .on('hideListing', function () {
                 var $fileSelector = this.getElement().find('.file-selector'),
                     $fileContainer = $fileSelector.find('.files'),
                     $placeholder = $fileSelector.find('.empty'),
                     $title = $fileSelector.find('.title'),
                     $uploader = $fileSelector.find('.file-upload-container'),
+                    $uploadForm = $uploader.parent('form'),
                     $switcher = $fileSelector.find('.upload-switcher a');
 
                 hider.show($switcher.filter('.listing'));
                 hider.hide($switcher.filter('.upload'));
-                $switcher.filter('.listing').css({display : 'inline-block'});
-
+                $switcher.filter('.listing').css({display: 'inline-block'});
+                $uploadForm.show();
                 hider.hide($fileContainer);
                 hider.hide($placeholder);
                 $title.text(__('Upload new custom interaction (zip package)'));
 
                 $uploader.uploader('reset');
                 hider.show($uploader);
             })
-            .on('updateListing', function(){
+            .on('updateListing', function () {
                 var self = this,
-                    urls = _.pick(this.config, ['disableUrl', 'enableUrl']),
+                    urls = _.pick(this.config, ['disableUrl', 'enableUrl', 'unregisterUrl', 'exportPciUrl']),
                     $fileSelector = this.getElement().find('.file-selector'),
                     $fileContainer = $fileSelector.find('.files'),
                     $placeholder = $fileSelector.find('.empty');
-
-                if(_.size(listing)){
+                if (_.size(listing)) {
 
                     hider.hide($placeholder);
 
                     $fileContainer
                         .empty()
                         .html(listingTpl({
-                            interactions : listing
+                            interactions: listing
                         }));
-
-                    $fileContainer.find('.switch-box').each(function(){
-                        var $switch = $(this);
-                        var $li = $switch.closest('.pci-list-element');
-                        var typeIdentifier = $li.data('type-identifier');
-                        switchFactory($switch, {
-                            on : {
-                                active : !$li.hasClass('pci-disabled')
+                    $fileContainer.find('.actions').each(function () {
+                        var pciswitch = $(this).find('.pci-switch');
+                        var pciUnregisterButton = $(this).find('.pci-unregister-button');
+                        var pciDownloadButton = $(this).find('.pci-download-button');
+                        var $li = $(this).closest('li');
+                        var typeIdentifier = $li.data('typeIdentifier');
+                        switchFactory(pciswitch, {
```

Comment:
Since the switch and the download button are only going to be used for certain PCIs, we should avoid running this code by wrapping those parts in something like `if (!listing[typeIdentifier].runtimeOnly) {}`

The delete button can of course stay outside the `if`.